cogboard-app/src/main/kotlin/com/cognifide/cogboard/http/HttpClient.kt

```diff
@@ -88,12 +89,9 @@ class HttpClient : AbstractVerticle() {
         val pass = config.getString(CogboardConstants.PROP_PASSWORD) ?: ""
         val token = config.getString(CogboardConstants.PROP_TOKEN) ?: ""
         val headers = config.getJsonObject(CogboardConstants.PROP_HEADERS)
+        val widgetType = config.getString(CogboardConstants.PROP_WIDGET_TYPE) ?: ""
 
-        if (user.isNotBlank() && token.isNotBlank()) {
-            request.basicAuthentication(user, token)
-        } else if (user.isNotBlank() && pass.isNotBlank()) {
-            request.basicAuthentication(user, pass)
-        }
+        AuthenticationFactory(user, pass, token, request).create(widgetType)
```

Comment:
This change is unacceptable because:

- any dev will have to know that there is a httpClient which uses some AuthenticationFactory in which he has to add his widget in order for it to work
- HttpClient shouldn't be aware that widgets even exist - it should only make requests for us